src/Atmos/Model/remainder.jl

```diff
@@ -99,9 +99,51 @@ compute_gradient_flux!(
 ) = compute_gradient_flux!(rem.main, diffusive, ∇transform, state, aux, t)
 
 function wavespeed(rem::RemainderModel, nM, state::Vars, aux::Vars, t::Real)
-    ref = aux.ref_state
-    return wavespeed(rem.main, nM, state, aux, t) -
-           sum(sub -> wavespeed(sub, nM, state, aux, t), rem.subs)
+    FT = eltype(state)
+
+    ws = fill(0, MVector{number_state_conservative(rem.main, FT), FT})
+    ws .= wavespeed(rem.main, nM, state, aux, t)
+
+    # I would like to write the following code but it doesn't compile on the
+    # GPU.
+    #
+    #     rs = fill(0, MVector{number_state_conservative(rem.main, FT), FT})
+    #     for sub in rem.subs
+    #         num_state = number_state_conservative(sub, FT)
+    #         rs[1:num_state] .+= wavespeed(sub, nM, state, aux, t)
+    #     end
+    #     ws .-= rs
+    #
+    # So I tried the following but it didn't work either.
+    #
+    #     @inbounds ntuple(Val(length(rem.subs))) do k
+    #         Base.@_inline_meta
+    #         @inbounds begin
+    #             sub = rem.subs[k]
+    #             num_state = number_state_conservative(sub, FT)
+    #              rs[1:num_state] .= rs[1:num_state] .+
+    #                                 wavespeed(sub, nM, state, aux, t)
+    #         end
+    #         nothing
+    #     end
+    #     ws .-= rs
+    #
+    # This works but I have the feeling it will stop working once there is more
+    # that one mode in `rem.subs`.
+    if !isempty(rem.subs)
+        sub = first(rem.subs)
+        rs = fill(0, MVector{number_state_conservative(rem.main, FT), FT})
+        rs[1:number_state_conservative(sub, FT)] .=
+            wavespeed(sub, nM, state, aux, t)
+        for sub in Base.tail(rem.subs)
+            rs[1:number_state_conservative(sub, FT)] .=
+                rs[1:number_state_conservative(sub, FT)] .+
+                wavespeed(sub, nM, state, aux, t)
+        end
+        ws .-= rs
+    end
+
+    return ws
```

Comment:
@vchuravy and @jkozdon: Do you guys have a suggestion on how to clean this code up?  I am struggling to get a clean version of the loop to compile on the GPU.

Reply:
Thanks to @jkozdon and StaticNumbers.jl I think I have a fix